tracer/tracer.go

```diff
@@ -2,15 +2,36 @@ package tracer
 
 import (
 	"log"
+	"os"
+	"strings"
 	"time"
 
 	"context"
 )
 
 const (
-	flushInterval = 2 * time.Second
+	defaultTraceEnabled = true
+	traceEnabledVar     = "DATADOG_TRACE_ENABLED"
+	flushInterval       = 2 * time.Second
 )
 
+var (
+	traceEnabled = defaultTraceEnabled
+)
+
+// Initialize globals based on environment variables.
+// Moved out of init() and accepting a function argument so it can be tested.
+func initTracerVars(lookup func(key string) (string, bool)) {
```

Comment:
I'm not really sure to use a generic lookup function here, since I find easier and without overhead to use just the ``os`` package. There is a particular reason / use case that I'm not considering to use just ``os.Getenv(traceEnabledVar)``?

Anyway, I have a couple of ideas to combine your PR with the #23! Will provide more details later.